Model/CardRepository.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Paytrail\PaymentService\Model;
+
+use Magento\Authorization\Model\UserContextInterface;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Vault\Api\PaymentTokenRepositoryInterface;
+use Paytrail\PaymentService\Api\CardRepositoryInterface;
+use Paytrail\PaymentService\Helper\ApiData;
+
+class CardRepository implements CardRepositoryInterface
+{
+    /**
+     * @var ApiData
+     */
+    private ApiData $apiData;
+
+    /**
+     * @var PaymentTokenRepositoryInterface
+     */
+    private PaymentTokenRepositoryInterface $paymentTokenRepository;
+
+    /**
+     * @var UserContextInterface
+     */
+    private UserContextInterface $userContext;
+
+    /**
+     * @param ApiData $apiData
+     * @param PaymentTokenRepositoryInterface $paymentTokenRepository
+     * @param UserContextInterface $userContext
+     */
+    public function __construct(
+        ApiData $apiData,
+        PaymentTokenRepositoryInterface $paymentTokenRepository,
+        UserContextInterface $userContext
+    ) {
+        $this->apiData = $apiData;
+        $this->paymentTokenRepository = $paymentTokenRepository;
+        $this->userContext = $userContext;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function save(): string
+    {
+        $response = $this->apiData->processApiRequest('add_card');
+
+        if (isset($response['error'])) {
+            throw new LocalizedException(__($response['error']), null, 403);
```

Comment:
Try to avoid translation declarations such as these. They will not work reliably. in this case the code responsible for collecting translatable strings will never pick this up. If the underlying code has a translated exception message, use that instead of declaring the translation again.